controllers/upgrade_handlers.go

```diff
@@ -158,47 +135,26 @@ func (r *ImageBasedUpgradeReconciler) rebootToNewStateRoot() error {
 // prePivot all the funcs needed to be called before a pivot
 func (r *ImageBasedUpgradeReconciler) prePivot(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
 	// backup with OADP
-	r.Log.Info("Backup with Oadp operator")
-	reqBackup, backupCRs, err := r.BackupRestore.CheckIfBackupRequested(ctx, ibu.Spec.OADPContent)
+	r.Log.Info("Handling backups with OADP operator")
+	ctrlResult, err := r.handleBackup(ctx, ibu)
 	if err != nil {
-		if k8serrors.IsInvalid(err) {
+		if backuprestore.IsBRFailedError(err) {
+			upgradeFailedStatus(ibu, err.Error())
+			return doNotRequeue(), nil
+		}
+		if backuprestore.IsBRFailedValidationError(err) || backuprestore.IsBRNotFoundError(err) {
 			utils.SetStatusCondition(&ibu.Status.Conditions,
```

Comment:
I wonder if we can create ready templates for each status and not to provide full description each time?
It just very long code each time when we set status, making code very difficult to read 
Not relevant to this PR i assume